runtime/src/extensions.rs

```diff
@@ -182,6 +194,52 @@ where
 	})
 }
 
+fn read_nfts_state<T, E>(
+	key: NftsKeys,
+	env: &mut Environment<E, BufInBufOutState>,
+) -> Result<Vec<u8>, DispatchError>
+where
+	T: pallet_contracts::Config + pallet_nfts::Config<CollectionId = CollectionId, ItemId = ItemId>,
+	E: Ext<T = T>,
+{
+	match key {
+		NftsKeys::Collection(collection) => {
+			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
+			Ok(pallet_nfts::Collection::<T>::get(collection).encode())
+		},
+		NftsKeys::CollectionOwner(collection) => {
+			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
+			Ok(pallet_nfts::Pallet::<T>::collection_owner(collection).encode())
+		},
+		NftsKeys::Item(collection, item) => {
+			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
+			Ok(pallet_nfts::Item::<T>::get(collection, item).encode())
+		},
+		NftsKeys::Owner(collection, item) => {
+			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
+			Ok(pallet_nfts::Pallet::<T>::owner(collection, item).encode())
+		},
+		NftsKeys::Attribute(collection, item, key) => {
+			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
+			Ok(pallet_nfts::Pallet::<T>::attribute(&collection, &item, &key).encode())
+		},
+		// NftsKeys::CustomAttribute(account, collection, item, key) => {
+		// 	env.charge_weight(T::DbWeight::get().reads(1_u64))?;
+		// 	Ok(pallet_nfts::Pallet::<T>::custom_attribute(&account, &collection, &item, &key)
```

Comment:
If this is commented out because `item` needs to be of type `Option<&ItemId>` you can use `item.as_ref()` to convert from a `Option<ItemId>` to `Option<&ItemId>`

Reply:
As above, see comment in description about `AccountId` issues.